lrpar/src/lib/diagnostics.rs

```diff
@@ -19,6 +19,17 @@ pub struct SpannedDiagnosticFormatter<'a> {
     nlc: OnceCell<NewlineCache>,
 }
 
+trait PushLine {
+    fn push_line<S: AsRef<str>>(&mut self, s: S);
+}
+
+impl PushLine for String {
+    fn push_line<S: AsRef<str>>(&mut self, s: S) {
```

Comment:
How about we call this `pushln` to mirror `println`? Also, a quick docstring on the trait to make that clear would be great.